backend/app/services/metrics_collector.py

```diff
@@ -0,0 +1,95 @@
+# Copyright (C) 2025 Intel Corporation
+# SPDX-License-Identifier: Apache-2.0
+
+import logging
+import time
+from datetime import UTC, datetime
+from multiprocessing import Lock, shared_memory
+from typing import NamedTuple
+from uuid import UUID
+
+import numpy as np
+
+from app.utils import Singleton
+
+logger = logging.getLogger(__name__)
+
+
+class LatencyMeasurement(NamedTuple):
+    """Individual latency measurement"""
+
+    model_id: str  # UUID as 36 character string "00000000-0000-0000-0000-000000000000"
+    latency_ms: float
+    timestamp: float
+
+
+class MetricsCollector(metaclass=Singleton):
```

Comment:
We are working to minimize the use of the Singleton pattern (see related issues [here](#4569)). Since metrics are collected within the same process, we can consider separating metric collection from retrieval. Probably, the `PipelineService` can access shared memory for metrics directly and collector don't need to be a Singleton in that case?